packages/ecs/src/topic.ts

```diff
@@ -0,0 +1,52 @@
+export type Topic<T extends string = string, D = any> = {
+  /**
+   * A property for the name of the topic
+   */
+  name: string
+
+  /**
+   * Provides iterator syntax to the consumers of a topic. This will loop
+   * through all of the events with the type specified in the second type 
+   * parameter. 
+   */
+  [Symbol.iterator](): IterableIterator<D>;
+
+  /**
+   * Pushes an event to the topic so that it can be consumed by downstream 
+   * systems.
+   */
+  pushEvent(event: D): void
+
+  /**
+   * Utility method that cleans the event list in the topic such that at the 
+   * end of the tick (or after all the systems have consumed whatever events
+   * they are interested in we can flush out events that have already been 
+   * consumed.)
+   */
+  flush(): void
+}
+
+/**
+ * The utility method used to create a topic, consumes a type parameter 
+ * and a name and returns an object that conforms to the topic type
+ */
+export const createTopic = <E>(name: string): Topic<string, E> => {
+ 
+  let events: E[] = [];
```

Comment:
I had some concerns about making this `let` vs `const`. I suppose it depends on whether or not we want to have control over how the array gets cleared out once all of the consumers have finished going through the topic. If we use `let` we get easier syntax (just reassign to a new array), but this means that we rely on the garbage collector to clean up after us. If we use `const` if we wanted to clean the array we would have to do something like `events.splice(0, events.length)`, which seems a little unnatural.